tests/meta/deliver/test_delivery_api.py

```diff
@@ -0,0 +1,111 @@
+"""Tests for delivery API"""
+
+
+def test_get_post_analysis_files(analysis_family, deliver_api):
+
+    # GIVEN a case which exist as bundle in hk
+    # GIVEN corresponding houskeeper tags for those files
+    case = analysis_family["internal_id"]
+    deliver_tags = ["case-tag"]
+
+    # WHEN we call get_post_analysis_files with matching case and file-tags and version is no version
+    version = None
+    bundle_latest_files = deliver_api.get_post_analysis_files(
+        case, version, deliver_tags
+    )
+
+    ## THEN housekeeper files should be returned
+    assert bundle_latest_files
+
+
+def test_get_post_analysis_files_version(analysis_family, deliver_api):
+
+    # GIVEN a case which exist as bundle in hk
+    # GIVEN corresponding houskeeper tags for those files
+    case = analysis_family["internal_id"]
+    deliver_tags = ["case-tag"]
+
+    # WHEN we call get_post_analysis_files with matching case and file-tags and version is supplied
+    version = "a_date"
+
+    bundle_version_files = deliver_api.get_post_analysis_files(
+        case, version, deliver_tags
+    )
+
+    ## THEN housekeeper files for those tags with the version should be returned
+    assert bundle_version_files
+
+
+def test_get_post_analysis_case_files(analysis_family, deliver_api):
+
+    # GIVEN a case which exist as bundle in hk
+    # GIVEN corresponding houskeeper tags for those files
+    case = analysis_family["internal_id"]
+    deliver_tags = ["case-tag"]
+
+    # WHEN we call get_post_analysis_case_files with matching case and file-tags and version
+    version = None
+    case_files = deliver_api.get_post_analysis_case_files(case, version, deliver_tags)
+
+    # THEN we should get those matching files back
+    assert case_files
+
+
+def test_get_post_analysis_case_files_only_sample_tags(analysis_family, deliver_api):
```

Comment:
This test and the previous test has the same GIVEN-WHEN-THEN. The normal suggestion in such a case is to remove one of the tests. I think we could update the GIVEN-WHEN-THENs so it clear what tests are testing ?

Reply:
Added a more clear comment